core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -393,6 +402,7 @@ class BlockManagerMasterEndpoint(
       topologyMapper.getTopologyForHost(idWithoutTopologyInfo.host))
 
     val time = System.currentTimeMillis()
+    executorIdToLocalDirs.get(id.executorId, () => localDirs)
```

Comment:
Is this call to populate the cache? Why not use `put`?

Reply:
Javadoc of `put` suggest to use the `get` method when the pattern is:  "if cached, return; otherwise create, cache and return": 
```java
/**
   * Associates {@code value} with {@code key} in this cache. If the cache previously contained a
   * value associated with {@code key}, the old value is replaced by {@code value}.
   *
   * <p>Prefer {@link #get(Object, Callable)} when using the conventional "if cached, return;
   * otherwise create, cache and return" pattern.
   *
   * @since 11.0
   */
  void put(K key, V value);
```

Reply:
But you are not using that pattern. You'd just adding something to the cache.

Reply:
True!